### PR TITLE
chore: adding additional logs to get original request in case of failure

### PR DESCRIPTION
**Pull request by saxenakshitiz**

### CODE REVIEW

File: hypertrace-graphql-entity-schema/src/main/java/org/hypertrace/graphql/entity/dao/GatewayServiceEntityDao.java [hunk 1]
```diff
@@ -84,7 +84,8 @@ public Single<EntityResultSet> getEntities(EntityRequest request) {
     return this.requestBuilder
         .buildRequest(request)
         .subscribeOn(this.boundedIoScheduler)
-        .flatMap(serverRequest -> this.fetchAndMapEntities(context, request, serverRequest));
+        .flatMap(serverRequest -> this.fetchAndMapEntities(context, request, serverRequest))
+        .doOnError(error -> log.error("Error while handling entities request {}", request, error));
```
Comment (aaron-steinfeld) [inline hypertrace-graphql-entity-schema/src/main/java/org/hypertrace/graphql/entity/dao/GatewayServiceEntityDao.java:88]: Have GW handle this (if it throws, it should include the details of what failed as well as logging it locally). At this layer, the only relevant logging would be a debug log on a successful received response.

Comment (saxenakshitiz) [reply hypertrace-graphql-entity-schema/src/main/java/org/hypertrace/graphql/entity/dao/GatewayServiceEntityDao.java:88]: GW logs the request. However I am looking for original request received by the graphql.

Comment (aaron-steinfeld) [reply hypertrace-graphql-entity-schema/src/main/java/org/hypertrace/graphql/entity/dao/GatewayServiceEntityDao.java:88]: That makes sense to log, but this isn't it - this is already an intermediate state, the result of the gql request processing. Ideally we should have the global gql error handler do such logging (we may need to do a bit of work to get to the initial request there).

Comment (saxenakshitiz) [reply hypertrace-graphql-entity-schema/src/main/java/org/hypertrace/graphql/entity/dao/GatewayServiceEntityDao.java:88]: Sure. Best to have a global error handler that logs original request on failure.

**Pull request closed without merging**